core/os/os.odin

```diff
@@ -206,11 +206,20 @@ heap_allocator_proc :: proc(allocator_data: rawptr, mode: mem.Allocator_Mode,
 		}
 	}
 
-	aligned_resize :: proc(p: rawptr, old_size: int, new_size: int, new_alignment: int) -> ([]byte, mem.Allocator_Error) {
+	aligned_resize :: proc(p: rawptr, old_size: int, new_size: int, new_alignment: int) -> (new_memory: []byte, err: mem.Allocator_Error) {
 		if p == nil {
 			return nil, nil
 		}
-		return aligned_alloc(new_size, new_alignment, p)
+
+		new_memory = aligned_alloc(new_size, new_alignment, p) or_return
```

Comment:
Wouldn't it be better to correct `heap_resize`?

Reply:
That's actually a good point.

Reply:
> Wouldn't it be better to correct `heap_resize`?
Actually, it can't be fixed there because `heap_resize` has no clue what the old size is, so it doesn't know what to zero.

This patch does appear to be the best compromise.